From: Richard Hughes <richard@hughsie.com>
Date: Thu, 19 Oct 2017 13:43:22 +0100
Subject: [PATCH] 0.116: Fix a critical warning on calling
 polkit_permission_new_sync with no system bus

Origin: upstream, 0.116, commit:984d16e6d21c6d6b0fc28d4fe7fe82575a43c95b

Imported from policykit-1_0.105-31.1.debian.tar.xz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/polkit/polkitpermission.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/polkit/polkitpermission.c b/src/polkit/polkitpermission.c
index f264094db206..d4b245915bd3 100644
--- a/src/polkit/polkitpermission.c
+++ b/src/polkit/polkitpermission.c
@@ -137,10 +137,13 @@ polkit_permission_finalize (GObject *object)
   g_free (permission->tmp_authz_id);
   g_object_unref (permission->subject);
 
-  g_signal_handlers_disconnect_by_func (permission->authority,
-                                        on_authority_changed,
-                                        permission);
-  g_object_unref (permission->authority);
+  if (permission->authority != NULL)
+    {
+      g_signal_handlers_disconnect_by_func (permission->authority,
+                                            on_authority_changed,
+                                            permission);
+      g_object_unref (permission->authority);
+    }
 
   if (G_OBJECT_CLASS (polkit_permission_parent_class)->finalize != NULL)
     G_OBJECT_CLASS (polkit_permission_parent_class)->finalize (object);
